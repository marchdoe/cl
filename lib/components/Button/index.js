import React from 'react'
import PropTypes from 'prop-types'

import styled from 'react-emotion'
import {
  border,
  color } from '../../theme'

const StyledButton = styled.button`
  display: inline-flex;
  font-family: 'helvetica';
  align-items: center;
  white-space: nowrap;
  font-size: 16px;
  border: 0.0625em solid ${color.gray4};
  height: 2.5em;
  justify-content: center;
  text-decoration: none;
  cursor: pointer;
  appearance: none;
  padding: 0 1em;
  border-radius: ${border.radius};
  box-sizing: border-box;
  transition: background-color 250ms ease-out, color 250ms ease-out, border-color 250ms ease-out;
  background-color: ${color.white};
  color: ${color.gray4};

  &:hover,
  &:focus,
  &:active {
    background-color: ${color.gray0};
    border-color: ${color.gray6};
  }

  &:focus {
    outline: none;
 }
`

const Button = ({ children, onClick }) => (
  <StyledButton
    onClick={onClick}
  >
    {children}
  </StyledButton>
)

Button.propTypes = {
  children: PropTypes.string.isRequired,
  onClick: PropTypes.func
}

export default Button
