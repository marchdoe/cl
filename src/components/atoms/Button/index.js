import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import theme from 'styled-theming'

import { color } from '../../../theme'

const backgroundColor = theme.variants('mode', 'type', {
  primary: { light: color.primary, dark: 'darkblue' },
  secondary: { light: 'green', dark: 'darkgreen' },
  danger: { light: 'orange', dark: 'darkorange' }
})

const StyledButton = styled.button`
  display: inline-flex;
  font-family: 'helvetica';
  align-items: center;
  white-space: nowrap;
  font-size: 16px;
  border: 0.0625em solid #ccc;
  height: 2.5em;
  justify-content: center;
  text-decoration: none;
  cursor: pointer;
  appearance: none;
  padding: 0 1em;
  border-radius: 3px;
  box-sizing: border-box;
  transition: background-color 250ms ease-out, color 250ms ease-out, border-color 250ms ease-out;
  background-color: ${backgroundColor};
  color: ${color.white};

  &:hover,
  &:focus,
  &:active {
    background-color: #000;
    border-color: #ccc;
  }

  &:focus {
    outline: none;
  }
`

const Button = ({ children, onClick, onMouseDown, onTouchStart, type }) => (
  <StyledButton
    onClick={onClick}
    onMouseDown={onMouseDown}
    onTouchStart={onTouchStart}
    type={type}
  >
    {children}
  </StyledButton>
)

Button.propTypes = {
  children: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  onMouseDown: PropTypes.func,
  onTouchStart: PropTypes.func,
  type: PropTypes.oneOf(['primary', 'secondary', 'danger'])
}

Button.defaultProps = {
  type: 'primary'
}

export default Button
