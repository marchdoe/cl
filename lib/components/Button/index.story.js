import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { withInfo } from '@storybook/addon-info'
import { withKnobs, boolean, text, number } from '@storybook/addon-knobs'

import Button from '.'

const stories = storiesOf('Button', module)

// Add the `withKnobs` decorator to add knobs support to your stories.
// You can also configure `withKnobs` as a global decorator.
stories.addDecorator(withKnobs)

// storiesOf('Button', module)
//   .add('default',
//     withInfo('')(() => (
//       <Button onClick={action('onClick')}>
//         Test
//       </Button>
//       )
//     )
//   )

// Knobs for React props
stories.add('with a button',
  withInfo('')(() => (
    <Button disabled={boolean('Disabled', false)} onClick={action('onClick')}>
      {text('Label', 'Hello Button')}
    </Button>
  )
))

// Knobs as dynamic variables.
stories.add('as dynamic variables', () => {
  const buttonText = text('Purchase', 'Purchase')
  const count = number('Count', 3)

  const content = `${buttonText} ${count}`
  return (<Button>{content}</Button>)
})
