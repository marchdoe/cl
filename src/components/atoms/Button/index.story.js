import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { withInfo } from '@storybook/addon-info'
import { withKnobs, boolean, text } from '@storybook/addon-knobs'

import Button from '.'

const stories = storiesOf('Button', module)

stories.addDecorator(withKnobs)

stories.add('with a button',
  withInfo('')(() => (
    <div>
      <Button type='primary' disabled={boolean('Disabled', false)} onClick={action('onClick')}>
        {text('Label', 'Hello Button')}
      </Button>

      <Button type='secondary' disabled={boolean('Disabled', false)} onClick={action('onClick')}>
        {text('Label', 'Hello Button')}
      </Button>

      <Button type='danger' disabled={boolean('Disabled', false)} onClick={action('onClick')}>
        {text('Label', 'Hello Button')}
      </Button>
    </div>
  )
))
