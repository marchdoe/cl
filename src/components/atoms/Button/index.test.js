import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Button from '.'

test('it works', () => {
  const tree = renderer.create(<Button>Hello World</Button>).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('font-size', '16px')
})
