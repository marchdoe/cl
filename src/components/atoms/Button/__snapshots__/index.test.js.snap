// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'helvetica';
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  font-size: 16px;
  border: 0.0625em solid #bec4c8;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-decoration: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 0 1em;
  border-radius: 3px;
  box-sizing: border-box;
  -webkit-transition: background-color 250ms ease-out,color 250ms ease-out,border-color 250ms ease-out;
  transition: background-color 250ms ease-out,color 250ms ease-out,border-color 250ms ease-out;
  background-color: #ffffff;
  color: #bec4c8;
}

.c0:hover,
.c0:focus,
.c0:active {
  background-color: #f8f9f9;
  border-color: #97a1a7;
}

.c0:focus {
  outline: none;
}

<button
  className="c0"
  onClick={undefined}
  onMouseDown={undefined}
  onTouchStart={undefined}
>
  Hello World
</button>
`;
